src/tools/crossgen2/Common/Compiler/TypeExtensions.cs

```diff
@@ -54,7 +54,7 @@ public static DefType GetClosestDefType(this TypeDesc type)
         public static bool RequiresInstArg(this MethodDesc method)
         {
             return method.IsSharedByGenericInstantiations &&
-                (method.HasInstantiation || method.Signature.IsStatic || method.ImplementationType.IsValueType);
+                (method.HasInstantiation || method.Signature.IsStatic || method.ImplementationType.IsValueType || (method.OwningType.IsInterface && !method.IsAbstract));
```

Comment:
Could you switch from `OwningType` to `ImplementationType` in both spots? They're always the same in both R2R and CoreRT, but it's one of those things that will let the type system enable the kind of generic inlining that UTC can do in .NET Native (it's a concept from the NUTC type system).

Reply:
Done